### PR TITLE
COM-19916: Implement the Blog Post full view

### PR DESCRIPTION
**Pull request by clash82**
https://jira.ez.no/browse/COM-19916

Dekstop view:
fireshot capture 14 - psalm 91 - ez platform - http___ezplatform lh_blog_psalm-91

### CODE REVIEW

**Comment by clash82** [comment]
review please @kamilmusial @SylvainGuittard @damianz5 
please start from https://github.com/ezsystems/ezplatform-com/pull/32/commits/a6af2a15e6c349433f12467b224d3129034eec31

File: app/Resources/views/full/blog_post.html.twig [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            <div class="container">
+                <h1>
+                    {{ ez_content_name(content) }}
+                </h1>
+            </div>
+        </div>
+
+        <div class="container">
+            <div class="row blog-post-author">
+                {% set authorId = ez_field_value(content, 'author') %}
```
Comment (kamilmusial) [inline app/Resources/views/full/blog_post.html.twig]: Is this variable necessary?

File: app/Resources/views/parts/blog_post/author.html.twig [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<div class="blog-post-author-image">
+    {{ ez_render_field(content, 'profile_picture', {
+        'parameters': {
+            'alias': 'blog_post_author'
+        }
+    }) }}
+</div>
+<div class="blog-post-author-name">
```
Comment (kamilmusial) [inline app/Resources/views/parts/blog_post/author.html.twig]: nitpick: I'd add an extra line between `<div>'s, like in `app/Resources/views/full/blog_post.html.twig`

Comment (clash82) [inline app/Resources/views/parts/blog_post/author.html.twig]: Good point

File: web/assets/css/blog_post.css [hunk 3]
```diff
@@ -0,0 +1,81 @@
+.blog-post-header {
+    height: 380px;
+    background: #fff;
+    display: flex;
+    align-items: center;
+    justify-content:center;
```
Comment (SylvainGuittard) [inline web/assets/css/blog_post.css]: First time, I can say: "CS: spacing"

giphy

Comment (clash82) [inline web/assets/css/blog_post.css]: Nice one, Sylvain :-)))

File: app/Resources/views/full/blog_post.html.twig [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        <div class="container">
+            <div class="row blog-post-author">
+                {{ render_esi(controller('ez_content:viewAction', {
+                    'contentId': ez_field_value(content, 'author'),
+                    'viewType': 'blog_post'
+                })) }}
+
+                <div class="blog-post-date">
+                    {{ ez_field_value(content, 'published_date')|date('M d, Y') }}
```
Comment (clash82) [inline app/Resources/views/full/blog_post.html.twig]: Changed.

**Comment by clash82** [comment]
Fixed, waiting for final approval @kamilmusial @SylvainGuittard

File: app/Resources/views/parts/blog_post/author.html.twig [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    By <span>{{ ez_field_value(content, 'first_name') }} {{ ez_field_value(content, 'last_name') }}</span>
+</div>
+
+<div class="blog-post-author-bio">
+    {{ ez_render_field(content, 'bio') }}
+</div>
+
+<div class="blog-post-author-company">
+    {{ ez_render_field(content, 'company') }}
+</div>
```
Comment (SylvainGuittard) [inline app/Resources/views/parts/blog_post/author.html.twig]: Social media icons are missing in this PR.
- github
- twitter
- linkedin
- facebook
- google plus

Important note: github account was not listed in the fields for the blog post content type. It should be added.

File: app/Resources/views/full/blog_post.html.twig [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{% extends 'pagelayout.html.twig' %}
+
```
Comment (SylvainGuittard) [inline app/Resources/views/full/blog_post.html.twig]: I just realized that another feature is missing: the canonical url
```
{% block canonical %}
    <link rel="canonical" href=ez_field_value(content, 'canonical_url')>
{% endblock %}
```
and add in the `<head>`:
```
{% block canonical %}
{% endblock %}
```

File: app/Resources/views/parts/blog_post/author.html.twig (hunk 5 above)
Comment (clash82) [inline app/Resources/views/parts/blog_post/author.html.twig]: Ffff...

File: app/Resources/views/full/blog_post.html.twig (hunk 6 above)
Comment (clash82) [inline app/Resources/views/full/blog_post.html.twig]: Specification was missing :(

Comment (clash82) [inline app/Resources/views/full/blog_post.html.twig]: Added.

File: app/Resources/views/parts/blog_post/author.html.twig (hunk 5 above)
Comment (clash82) [inline app/Resources/views/parts/blog_post/author.html.twig]: Added.

File: app/Resources/views/full/blog_post.html.twig (hunk 6 above)
Comment (SylvainGuittard) [inline app/Resources/views/full/blog_post.html.twig]: Indeed. Sorry my bad.

**Pull request merged**